packages/space-opera/src/components/materials_panel/materials_panel.ts

```diff
@@ -137,50 +137,77 @@ export class MaterialPanel extends ConnectedLitElement {
     this.safeUrlIds = safeUrlIds;
   }
 
-  /* Interpolate from red to original color as n approaches maxN */
-  getInterpolatedColor(original: RGBA, n: number, maxN: number): RGBA {
-    const red: RGBA = [1, 0, 0, 1];
-    const percentRed = (maxN - n) / maxN;
-    const percentOriginal = 1 - percentRed;
+  /* Interpolate base color as curr approaches duration */
+  getInterpolatedColor(original: RGBA, curr: number, duration: number): RGBA {
+    const interpColor = [0, 0, 0];
+    // determine how much of interp color to use
+    const interpRatio = (duration - curr) / duration;
+    const originalRatio = 1 - interpRatio;
     return [
-      (percentRed * red[0]) + (percentOriginal * original[0]),
-      (percentRed * red[1]) + (percentOriginal * original[1]),
-      (percentRed * red[2]) + (percentOriginal * original[2]),
+      (interpRatio * interpColor[0]) + (originalRatio * original[0]),
+      (interpRatio * interpColor[1]) + (originalRatio * original[1]),
+      (interpRatio * interpColor[2]) + (originalRatio * original[2]),
       original[3],
     ];
   }
 
+  getInterpolatedEmissive(original: RGB, curr: number, duration: number): RGB {
```

Comment:
You could combine these two functions if you modified `original` instead of returning a new object. Also keeps you from allocating as many things. But if that has bad consequences elsewhere, feel free to ignore.

Reply:
going to keep as is, having the two different functions results in cleaner code elsewhere.